contracts/src/core/EigenDABlobVerifier.sol

```diff
@@ -0,0 +1,302 @@
+     * @param blobVerificationProof The blob verification proof for the blob
+     * @param nonSignerStakesAndSignature The nonSignerStakesAndSignature for the blob
+     * @param securityThreshold The custom security threshold to verify the blob against
+     * @param quorumNumbersRequired The required quorum numbers
+     */
+    function verifyBlobV2(
+        BatchHeaderV2 calldata batchHeader,
+        BlobVerificationProofV2 calldata blobVerificationProof,
+        NonSignerStakesAndSignature calldata nonSignerStakesAndSignature,
+        SecurityThresholds memory securityThreshold,
+        bytes calldata quorumNumbersRequired
+    ) external view {
+        EigenDABlobVerificationUtils._verifyBlobV2ForQuorums(
+            eigenDAThresholdRegistry,
+            eigenDASignatureVerifier,
+            batchHeader,
+            blobVerificationProof,
+            nonSignerStakesAndSignature,
+            securityThreshold,
+            quorumNumbersRequired
+        );
+    }
+
+    /**
+     * @notice Verifies a blob for a defined set of quorums and a set of custom security thresholds
+     * @param batchHeader The batch header of the blob
+     * @param blobVerificationProof The blob verification proof for the blob
+     * @param nonSignerStakesAndSignature The nonSignerStakesAndSignature for the blob
+     * @param securityThresholds The set of custom security thresholds to verify the blob against
+     * @param quorumNumbersRequired The required quorum numbers
+     */
+    function verifyBlobV2(
+        BatchHeaderV2 calldata batchHeader,
+        BlobVerificationProofV2 calldata blobVerificationProof,
+        NonSignerStakesAndSignature calldata nonSignerStakesAndSignature,
+        SecurityThresholds[] memory securityThresholds,
+        bytes calldata quorumNumbersRequired
+    ) external view {
+        EigenDABlobVerificationUtils._verifyBlobV2ForQuorumsForThresholds(
+            eigenDAThresholdRegistry,
+            eigenDASignatureVerifier,
+            batchHeader,
+            blobVerificationProof,
+            nonSignerStakesAndSignature,
+            securityThresholds,
+            quorumNumbersRequired
+        );
+    }
+
+    ///////////////////////// HELPER FUNCTIONS ///////////////////////////////
+
+    /**
+     * @notice Returns the nonSignerStakesAndSignature for a given blob and signed batch
+     * @param signedBatch The signed batch to get the nonSignerStakesAndSignature for
+     * @param blobHeader The blob header of the blob in the signed batch
+     */
+    function getNonSignerStakesAndSignature(
+        SignedBatch calldata signedBatch,
+        BlobHeaderV2 calldata blobHeader
+    ) external view returns (NonSignerStakesAndSignature memory) {
+        return EigenDABlobVerificationUtils._getNonSignerStakesAndSignature(
+            operatorStateRetriever, 
+            registryCoordinator, 
+            signedBatch.attestation, 
+            blobHeader.quorumNumbers
```

Comment:
this is apparently a bug, and we should use the quorumNumbers from the signedBatch. cc @mooselumph for details (I didnt understand why)